### ISSUE TITLE
Creating items in nested lists

### ISSUE THREAD

**Issue by KrzysztofMadejski** [akesterson/dpath-python#62]
Hi,

I would expect that running `dpath.util.new` on `row[0][field1]` would created a nested structure that would result in an `dict -> list -> dict`:
```
+ {'row': [{'field1': 'val1', 'field2': 'val2'}, {'field1': 'val3'}]}
?         ^                                                        +

instead of 

- {'row': {'0': {'field1': 'val1', 'field2': 'val2'}, '1': {'field1': 'val3'}}}
?         ^^^^^^                                     -----                    -
```

What are your thoughts? I'd love it. Note: It would break backward compatibility..

### Coding
Coding it seems quite easy:
https://github.com/akesterson/dpath-python/blob/d0380f817265ffd043a261eac8a3f5380f7577a0/dpath/path.py#L27
should be turn into
```python
 is_num = re.match('[\d]+', elem)
 result.append([elem, (list if is_num else dict)])
```
plus maybe accessors should be modified.

BTW. Thank you for this lib!

**Comment by deisi** [comment]
Aha I was wrapping my head around this trying to find out why this doesn't work, but I see.

**Comment by KrzysztofMadejski** [comment]
@akesterson what do you think about it?

**Comment by rbdixon** [comment]
I hit this one, too. Had to implement a ugly little workaround.

**Comment by akesterson** [comment]
Duplicate of #91, see the fix there and merged in pull request #104.
